In [1]:
import os
import sys
import pprint

base_dir = 'F:\Thesis_ssd\MasterThesis3.0'
os.chdir(base_dir)

from Classes.DataProcessing.LoadData import LoadData
from Classes.DataProcessing.DataHandler import DataHandler
from Classes.DataProcessing.DataGenerator import DataGenerator
from Classes.Modeling.GridSearchResultProcessor import GridSearchResultProcessor
from Classes.Modeling.CustomCallback import CustomCallback
from Classes.Scaling.ScalerFitter import ScalerFitter
from Classes.Scaling.MinMaxScalerFitter import MinMaxScalerFitter
from Classes.Scaling.StandardScalerFitter import StandardScalerFitter
from Classes.Modeling.ResultFitter import ResultFitter
import json

In [16]:
load_args = {
    'earth_explo_only' : True,
    'noise_earth_only' : False,
    'downsample' : True,
    'upsample' : True,
    'frac_diff' : 0.5,
    'seed' : 1,
    'subsample_size' : 0.25
}

loadData = LoadData(**load_args)

full_ds, train_ds, val_ds, test_ds = loadData.get_datasets()
noise_ds = loadData.noise_ds
handler = DataHandler(loadData)
dataGen = DataGenerator(loadData)
resultFitter = ResultFitter(loadData)

AttributeError: 'LoadData' object has no attribute 'getDatasets'

In [6]:
result_file_name = 'results_6_noiseNotNoise_timeAug_sscale_noiseAug_earlyS.csv'
model = resultFitter.fit_from_csv_and_index(result_file_name, 0, 2, 10, use_tensorboard = False, 
                               use_liveplots = True, use_custom_callback = False)

NameError: name 'resultFitter' is not defined

In [39]:
result_file_name = 'results_LSTM_NARROW_noiseNotNoise_timeAug_sscale_noiseAug_earlyS.csv'
df = GridSearchResultProcessor().get_results_df_by_name(result_file_name, 2)

In [40]:
df[df.columns[:15]]

batch_size  epochs  learning_rate  num_layers optimizer activation  \
0           64      33           0.01           2       sgd       tanh   
1           64      33           0.01           2       sgd       tanh   
2           64      33           0.01           2       sgd       tanh   
3           64      33           0.01           2       sgd       tanh   
4           64      33           0.01           2       sgd       tanh   
5           64      33           0.01           2       sgd       tanh   
6           64      33           0.01           2       sgd       tanh   
7           64      33           0.01           2       sgd       tanh   
8           64      33           0.01           2      adam       tanh   
9           64      33           0.01           2   rmsprop       tanh   
10          64      33           0.01           2       sgd       tanh   
11          64      33           0.01           2       sgd       tanh   
12          64      33           0.01           2       sgd       tanh   
13          64      33           0.01           2       sgd       tanh   
14          64      33           0.01           2       sgd       tanh   
15          64      33           0.01           2       sgd       tanh   
16          64      33           0.01           2       sgd       tanh   
17          64      33           0.01           2       sgd       tanh   
18          64      29           0.01           2       sgd       tanh   
19          64      37           0.01           2       sgd       tanh   
20          64      33           0.01           2       sgd       tanh   
21          64      33           0.01           2       sgd       tanh   
22          64      33           0.01           2       sgd       tanh   
23          64      33           0.01           2       sgd       tanh   
24          64      33           0.01           2       sgd       tanh   
25          64      33           0.01           2       sgd       tanh   
26          64      33           0.01           2       sgd       relu   
27          64      33           0.01           2       sgd    sigmoid   
28          64      33           0.01           2       sgd    softmax   
29          64      33           0.01           2       sgd       tanh   
30          64      33           0.01           2       sgd       tanh   

   decay_sequence  dropout_rate  filters  kernel_size    l1_r   l2_r  \
0          [1, 2]         0.500       17            5  0.0001  0.001   
1          [1, 2]         0.500       17            5  0.0001  0.001   
2          [1, 2]         0.400       17            5  0.0001  0.001   
3          [1, 2]         0.200       17            5  0.0001  0.001   
4          [1, 2]         0.100       17            5  0.0001  0.001   
5          [1, 2]         0.010       17            5  0.0001  0.001   
6          [1, 2]         0.001       17            5  0.0001  0.001   
7          [1, 2]         0.000       17            5  0.0001  0.001   
8          [1, 2]         0.300       17            5  0.0001  0.001   
9          [1, 2]         0.300       17            5  0.0001  0.001   
10         [1, 2]         0.300       17            5  0.0001  0.001   
11         [1, 2]         0.300       17            5  0.0001  0.001   
12         [1, 2]         0.300       17            5  0.0001  0.001   
13         [1, 2]         0.300       17            5  0.0001  0.001   
14         [1, 2]         0.300       17            5  0.0001  0.001   
15         [1, 2]         0.300       17            5  0.0001  0.001   
16         [1, 2]         0.300       17            5  0.0001  0.001   
17         [1, 2]         0.300       17            5  0.0001  0.001   
18         [1, 2]         0.300       17            5  0.0001  0.001   
19         [1, 2]         0.300       17            5  0.0001  0.001   
20         [1, 2]         0.300       17            5  0.3000  0.001   
21         [1, 2]         0.300       17           

In [41]:
df[df.columns[13:]]

padding  start_neurons  train_loss  train_accuracy  train_precision  \
0     same             32    0.843425        0.928650         0.862717   
1     same             32    0.856955        0.949212         0.845839   
2     same             32    0.832348        0.944839         0.879387   
3     same             32    0.759288        0.980216         0.920956   
4     same             32    0.765186        0.966016         0.935662   
5     same             32    0.782132        0.983592         0.944435   
6     same             32    0.793265        0.982209         0.929980   
7     same             32    1.044652        0.971432         0.922752   
8     same             32    2.342889        0.822314         0.720338   
9     same             32    1.584430        0.763097         0.711021   
10    same             32    0.796258        0.968050         0.905414   
11    same             16    0.559328        0.946777         0.902908   
12    same             32    0.775099        0.972447         0.907002   
13    same             64    1.184011        0.948904         0.915775   
14    same            128    1.946189        0.955922         0.913227   
15    same            256    3.325231        0.934577         0.905498   
16    same            512    5.499685        0.900792         0.873956   
17    same           1024    8.477287        0.859407         0.831843   
18    same             32    0.869004        0.954148         0.901362   
19    same             32    0.710379        0.951974         0.920872   
20    same             32  343.871674        0.658513         0.658130   
21    same             32  155.409958        0.658629         0.658506   
22    same             32   39.233425        0.658631         0.658631   
23    same             32    1.057339        0.658882         0.658882   
24    same             32    1.321735        0.698032         0.708681   
25    same             32    0.456250        0.971378         0.921750   
26    same             32    2.165133        0.658368         0.657670   
27    same             32    1.024562        0.658924         0.658924   
28    same             32    1.023060        0.658840         0.658840   
29    same             32    0.774593        0.961107         0.907754   
30    same             32         NaN             NaN              NaN   

    train_recall    val_loss  val_accuracy  val_precision  val_recall  
0       0.857514    0.970667      0.868316       0.775568    0.778812  
1       0.809346    0.980064      0.903571       0.766572    0.724409  
2       0.867630    1.003364      0.888981       0.783144    0.768074  
3       0.898200    1.083491      0.900166       0.794034    0.774517  
4       0.935273    1.217005      0.874209       0.786458    0.790981  
5       0.931207    1.338903      0.889803       0.787405    0.774517  
6       0.910209    1.372538      0.886949       0.779356    0.763780  
7       0.909552    1.513449      0.885833       0.776989    0.760916  
8       0.734196    2.349560      0.814815       0.708807    0.724409  
9       0.813967    1.595288      0.757818       0.705492    0.815319  
10      0.885676    1.000532      0.909462       0.813920    0.798139  
11      0.903436    0.761700      0.874040       0.799716    0.814603  
12      0.883908    1.013919      0.909168       0.794981    0.766643  
13      0.921833    1.459759      0.866363       0.795455    0.816750  
14      0.910284    2.231662      0.885600       0.794981    0.792412  
15      0.921068    3.695782      0.846094       0.758049    0.775233  
16      0.908843    5.866838      0.819820       0.741951    0.781675  
17      0.890516    8.841166      0.801769       0.726326    0.778812  
18      0.893229    1.062029      0.881452       0.786458    0.782391  
19      0.926614    0.963329      0.881151       0.815341    0.833214  
20      0.998985  343.869812      0.661458       0.661458    1.000000  
21      0.999683  155.408386      0.661458       0

## Findings:
 - New optimal dropout_rate: 0.2 (train_acc: 98%, val_acc: 90%) Old 0.3 (train_acc: 96.8%, val_acc: 90.9%)
 - RELU and Sigmoid activation functions do not work with this model.
 - L1 regularization is counter productive (Less is more) so let l1 = 0.00
 - 32 start neurons appear to be optimal

In [ ]:
li = fit_from_csv_and_index(result_file_name, 0, 2)

In [ ]:
mysplit(result_fi)

In [60]:
class A(B):
    def __init__(self, hey):
        self.hey = hey

class B:
    def __init__(self):
        pass
    def check(self):
        print(hasattr(self, 'hey'))

In [61]:
A = A("bye")

In [62]:
A.check()

AttributeError: 'A' object has no attribute 'check'

In [58]:
hasattr(A, 'hey')

True